src/libs/ReportUtils.js

```diff
@@ -455,6 +455,11 @@ function isAllowedToComment(report) {
         return true;
     }
 
+    // If user opens public chat room directly from deep link after logout
+    if (!allPolicies) {
+        return false;
+    }
+
     // If we've made it here, commenting on this report is restricted.
     // If the user is an admin, allow them to post.
     const policy = allPolicies[`${ONYXKEYS.COLLECTION.POLICY}${report.policyID}`];
```

Comment:
We could also do this, right?
```suggestion
    const policy = allPolicies && allPolicies[`${ONYXKEYS.COLLECTION.POLICY}${report.policyID}`];
```

Reply:
right so you want this change and remove this early return?
```
    if (!allPolicies) {
        return false;
    }
```
Adding both is redundant.
For commenting purpose, I'd suggest earlyreturn

Reply:
Lets leave an early return, its fine